--- tuxmath.desktop.orig	2010-12-03 00:36:43.716020606 +0200
+++ tuxmath.desktop	2010-12-03 00:37:03.855015886 +0200
@@ -4,11 +4,13 @@
 GenericName[de]=Mathe Spiel
 GenericName[ru]=Образовательная игра
 GenericName[fr]=Jeu mathématique
+GenericName[tr]=Eğitsel matematik oyunu
 Comment=Tux Math - Learn math with Tux!
 Comment[pl]=Tux Math - Ucz siê matematyki z Tuxem!
 Comment[de]=Tux Math - Lerne Mathe mit Tux!
 Comment[ru]=Обучение арифметике с пингвином в главной роли
 Comment[fr]=Tux Math - Apprenez les mathématiques avec Tux!
+Comment[tr]=TuxMath Matematik Oyunu
 Exec=tuxmath
 Icon=tuxmath
 Type=Application
