--- C4/Auth_with_ldap.pm.orig	2011-11-19 11:15:11.605000893 +0200
+++ C4/Auth_with_ldap.pm	2011-11-19 11:23:15.683998819 +0200
@@ -136,10 +136,11 @@
 		}
         my $search = search_method($db, $userid) or return 0;   # warnings are in the sub
         $userldapentry = $search->shift_entry;
-		my $cmpmesg = $db->compare( $userldapentry, attr=>'userpassword', value => $password );
-		if ($cmpmesg->code != 6) {
-			warn "LDAP Auth rejected : invalid password for user '$userid'. " . description($cmpmesg);
-			return 0;
+        my $dn = $userldapentry->dn;
+        my $user_ldap_bind_ret = $db->bind($dn, password => $password); 
+        if ($user_ldap_bind_ret->code) {
+           warn "LDAP Auth rejected : invalid password for user '$userid'. " . description($user_ldap_bind_ret);
+           return 0;
 		}
 	}
 
