--- src/ktechlab.desktop.orig	2008-07-15 12:14:30.000000000 +0300
+++ src/ktechlab.desktop	2008-07-15 12:16:17.000000000 +0300
@@ -5,8 +5,10 @@
 Icon=ktechlab
 Type=Application
 DocPath=ktechlab/index.html
+GenericName=IDE for microcontrollers and electronics
+GenericName[tr]=Mikrodenetleyici ve elektronik geliştirme ortamı
 Comment=An IDE for microcontrollers and electronics
 Comment[fr]=Une interface de développement intégrée pour microcontrôleurs et électronique
 Terminal=false
 MimeType=application/x-ktechlab;application/x-flowcode;application/x-circuit;application/x-microbe;
-Categories=Qt;KDE;Education;Science;Electronics
+Categories=Application;Science;Engineering;Electronics
