--- configure.orig	2010-12-30 11:23:55.801698461 +0200
+++ configure	2010-12-30 11:25:34.193573291 +0200
@@ -4249,7 +4249,7 @@
 
 if test "x$LUAPFX" != x ; then
     LUA_CFLAGS="-I$LUAPFX"
-    LUA_LIBS="-L$LUAPFX -llua -llualib"
+    LUA_LIBS="-L$LUAPFX -llua"
 else
     for ac_prog in lua-config lua-config50 lua-config5.0
 do
@@ -4484,13 +4484,13 @@
 fi
 
 
-            { $as_echo "$as_me:${as_lineno-$LINENO}: checking for luaopen_base in -llualib" >&5
-$as_echo_n "checking for luaopen_base in -llualib... " >&6; }
+            { $as_echo "$as_me:${as_lineno-$LINENO}: checking for luaopen_base in -llua" >&5
+$as_echo_n "checking for luaopen_base in -llua... " >&6; }
 if test "${ac_cv_lib_lualib_luaopen_base+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-llualib $LUA_LIBS $LIBS"
+LIBS="-llua $LUA_LIBS $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -4521,7 +4521,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_lualib_luaopen_base" >&5
 $as_echo "$ac_cv_lib_lualib_luaopen_base" >&6; }
 if test "x$ac_cv_lib_lualib_luaopen_base" = x""yes; then :
-  LUA_LIBS="$LUA_LIBS -llualib"
+  LUA_LIBS="$LUA_LIBS -llua"
 else
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for luaopen_base in -llualib50" >&5
 $as_echo_n "checking for luaopen_base in -llualib50... " >&6; }
