--- configure.orig	2010-01-15 21:22:42.089557891 +0200
+++ configure	2010-01-15 21:25:30.323555624 +0200
@@ -4499,7 +4499,7 @@
 
 if test "x$LUAPFX" != x ; then
     LUA_CFLAGS="-I$LUAPFX"
-    LUA_LIBS="-L$LUAPFX -llua -llualib"
+    LUA_LIBS="-L$LUAPFX -llua"
 else
     for ac_prog in lua-config lua-config50 lua-config5.0
 do
@@ -4829,13 +4829,13 @@
 fi
 
 
-            { $as_echo "$as_me:$LINENO: checking for luaopen_base in -llualib" >&5
-$as_echo_n "checking for luaopen_base in -llualib... " >&6; }
+            { $as_echo "$as_me:$LINENO: checking for luaopen_base in -llua" >&5
+$as_echo_n "checking for luaopen_base in -llua... " >&6; }
 if test "${ac_cv_lib_lualib_luaopen_base+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-llualib $LUA_LIBS $LIBS"
+LIBS="-llua $LUA_LIBS $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -4895,7 +4895,7 @@
 { $as_echo "$as_me:$LINENO: result: $ac_cv_lib_lualib_luaopen_base" >&5
 $as_echo "$ac_cv_lib_lualib_luaopen_base" >&6; }
 if test "x$ac_cv_lib_lualib_luaopen_base" = x""yes; then
-  LUA_LIBS="$LUA_LIBS -llualib"
+  LUA_LIBS="$LUA_LIBS -llua"
 else
   { $as_echo "$as_me:$LINENO: checking for luaopen_base in -llualib50" >&5
 $as_echo_n "checking for luaopen_base in -llualib50... " >&6; }
