--- makefile.orig	2009-07-27 15:34:40.000000000 +0300
+++ makefile	2009-07-27 15:35:02.000000000 +0300
@@ -18,7 +18,7 @@
 CXXFLAGS += `sdl-config --cflags` -DVERSION=$(VERSION) -DRELEASE=$(RELEASE) -DUSEPAK=$(USEPAK)
 CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DUNIX -DGAMEPLAYMANUAL=\"$(DOCDIR)index.html\" -Wall
 CXXFLAGS += -DLOCALEDIR=\"$(LOCALEDIR)\"
-CXXFLAGS += $(CFLAGS) -Werror
+CXXFLAGS += $(CFLAGS)
 LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lz
 
 OBJS += CAudio.o
